consensus/contracts.go

```diff
@@ -36,274 +36,152 @@ func (s *State) storageProofSegmentIndex(contractID ContractID, windowIndex Bloc
+		hash.CalculateSegments(contract.FileSize),
 		segmentIndex,
-		openContract.FileContract.FileMerkleRoot,
-	) {
+		contract.FileMerkleRoot,
+	)
+	if !verified {
 		return errors.New("provided storage proof is invalid")
 	}
 
 	return nil
 }
 
-// applyStorageProof takes a storage proof and adds any outputs created by it
-// to the consensus state.
-//
-// TODO: Though the contract terminates here, code later on handles that. That
-// should be changed.
-func (s *State) applyStorageProof(sp StorageProof, td *TransactionDiff) {
-	openContract := s.openContracts[sp.ContractID]
-	contractDiff := ContractDiff{
-		Contract:             openContract.FileContract,
-		ContractID:           sp.ContractID,
-		New:                  false,
-		PreviousOpenContract: *openContract,
-	}
-
-	// Set the payout of the output - payout cannot be greater than the
-	// amount of funds remaining.
-	payout := openContract.FileContract.ValidProofPayout
-	if openContract.FundsRemaining < openContract.FileContract.ValidProofPayout {
-		payout = openContract.FundsRemaining
-	}
-
-	// Create the output and add it to the list of unspent outputs.
-	output := Output{
-		Value:     payout,
-		SpendHash: openContract.FileContract.ValidProofAddress,
-	}
-	outputID, err := openContract.FileContract.StorageProofOutputID(openContract.ContractID, s.height(), true)
-	if err != nil {
-		panic(err)
+// addContract takes a FileContract and its corresponding ContractID and adds
+// it to the state.
+func (s *State) applyContract(contract FileContract, id ContractID) (cd ContractDiff) {
+	s.openContracts[id] = contract
+	cd = ContractDiff{
+		New:      true,
+		ID:       id,
+		Contract: contract,
 	}
-	s.unspentOutputs[outputID] = output
-	td.OutputDiffs = append(td.OutputDiffs, OutputDiff{New: true, ID: outputID, Output: output})
-
-	// Mark the proof as complete for this window, and subtract from the
-	// FundsRemaining.
-	s.openContracts[sp.ContractID].WindowSatisfied = true
-	s.openContracts[sp.ContractID].FundsRemaining -= payout
-	contractDiff.NewOpenContract = *s.openContracts[sp.ContractID]
 	return
 }
 
-func (s *State) invertStorageProof(sp StorageProof) (diff OutputDiff) {
-	openContract := s.openContracts[sp.ContractID]
-	outputID, err := openContract.FileContract.StorageProofOutputID(openContract.ContractID, s.height(), true)
-	if err != nil {
-		panic(err)
+// applyStorageProof takes a storage proof and adds any outputs created by it
+// to the consensus state.
+func (s *State) applyStorageProof(sp StorageProof) (od OutputDiff, cd ContractDiff) {
+	// Calculate the new output and its id.
+	contract := s.openContracts[sp.ContractID]
+	output := Output{
+		Value:     contract.Payout,
+		SpendHash: contract.ValidProofAddress,
 	}
-	output, err := s.output(outputID)
+	outputID, err := contract.StorageProofOutputID(sp.ContractID, s.height(), true)
 	if err != nil {
-		panic(err)
+		if DEBUG {
+			panic(err) // hard to avoid
+		}
```

Comment:
apparently none. StorageProofOutputID used to be a more complex function that had error conditions.

Now it seems that the error is unneeded, thanks to some simplifications.